### PR TITLE
GEODE-8269: Improve test coverage

### PR DESCRIPTION
**Pull request by sabbeyPivotal**
Analyzed test coverage via Intellij and by looking through all tests.  There were some test cases missing.  We either added them or created additional stories in Tracker.

### CODE REVIEW

File: geode-redis/src/test/java/org/apache/geode/redis/internal/executor/key/ExpireAtExecutorJUnitTest.java [hunk 1]
```diff
@@ -34,30 +37,30 @@
...
-    commandsAsBytesWithTooFewArguments.add("key".getBytes());
-    Command command = new Command(commandsAsBytesWithTooFewArguments);
+    List<byte[]> commandAsBytes = new ArrayList<>();
+    commandAsBytes.add("EXPIREAT".getBytes());
+    commandAsBytes.add("key".getBytes());
+    Command command = new Command(commandAsBytes);
 
-    RedisResponse response =
-        new ExpireAtExecutor().executeCommand(command, mockContext());
+    Throwable thrown = catchThrowable(() -> command.execute(mockContext()));
```
Comment (jdeppe-pivotal) [inline geode-redis/src/test/java/org/apache/geode/redis/internal/executor/key/ExpireAtExecutorJUnitTest.java:45]: Can this be a `assertThatThrownBy` assertion?

File: geode-redis/src/test/java/org/apache/geode/redis/internal/executor/key/ExpireAtExecutorJUnitTest.java [hunk 2]
```diff
@@ -34,30 +37,30 @@
...
+    commandsAsBytes.add("extra-argument".getBytes());
+    Command command = new Command(commandsAsBytes);
 
-    RedisResponse response =
-        new ExpireAtExecutor().executeCommand(command, mockContext());
+    Throwable thrown = catchThrowable(() -> command.execute(mockContext()));
 
-    assertThat(response.toString()).startsWith("-ERR The wrong number of arguments");
+    AssertionsForClassTypes.assertThat(thrown).hasMessageContaining("wrong number of arguments");
+    AssertionsForClassTypes.assertThat(thrown).isInstanceOf(RedisParametersMismatchException.class);
```
Comment (jdeppe-pivotal) [inline geode-redis/src/test/java/org/apache/geode/redis/internal/executor/key/ExpireAtExecutorJUnitTest.java:63]: Maybe the same thing, but at least for consistency, `assertThat` should come from the `org.assertj.core.api.Assertions` package.

File: geode-redis/src/test/java/org/apache/geode/redis/internal/executor/key/ExpireExecutorJUnitTest.java [hunk 3]
```diff
@@ -35,42 +37,41 @@
...
+    commandsAsBytes.add("key".getBytes());
+    Command command = new Command(commandsAsBytes);
 
-    Command command = new Command(commandsAsBytesWithTooFewArguments);
-    RedisResponse response = executor.executeCommand(command, mockContext());
+    Throwable thrown = catchThrowable(() -> command.execute(mockContext()));
 
-    assertThat(response.toString()).startsWith("-ERR The wrong number of arguments");
+    AssertionsForClassTypes.assertThat(thrown).hasMessageContaining("wrong number of arguments");
+    AssertionsForClassTypes.assertThat(thrown).isInstanceOf(RedisParametersMismatchException.class);
```
Comment (jdeppe-pivotal) [inline geode-redis/src/test/java/org/apache/geode/redis/internal/executor/key/ExpireExecutorJUnitTest.java:48]: Ditto previous comment.

File: geode-redis/src/test/java/org/apache/geode/redis/internal/executor/key/KeysExecutorJUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    List<byte[]> commandsAsBytes = new ArrayList<>();
+    commandsAsBytes.add("KEYS".getBytes());
+    commandsAsBytes.add("*".getBytes());
+    commandsAsBytes.add("**".getBytes());
+    Command command = new Command(commandsAsBytes);
+
+    Throwable thrown = catchThrowable(() -> command.execute(mockContext()));
+
+    AssertionsForClassTypes.assertThat(thrown).hasMessageContaining("wrong number of arguments");
+    AssertionsForClassTypes.assertThat(thrown).isInstanceOf(RedisParametersMismatchException.class);
```
Comment (jdeppe-pivotal) [inline geode-redis/src/test/java/org/apache/geode/redis/internal/executor/key/KeysExecutorJUnitTest.java:60]: Ditto previous comment

File: geode-redis/src/test/java/org/apache/geode/redis/internal/executor/key/ExpireAtExecutorJUnitTest.java (hunk 1 above)
Comment (sabbeyPivotal) [reply geode-redis/src/test/java/org/apache/geode/redis/internal/executor/key/ExpireAtExecutorJUnitTest.java:45]: yeah! good catch.

File: geode-redis/src/test/java/org/apache/geode/redis/internal/executor/key/ExpireAtExecutorJUnitTest.java (hunk 2 above)
Comment (sabbeyPivotal) [reply geode-redis/src/test/java/org/apache/geode/redis/internal/executor/key/ExpireAtExecutorJUnitTest.java:63]: Definitely. Editing those now.

**Comment by sabbeyPivotal** [comment]
AcceptanceTestOpenJDK11 failure is not a result of these code changes.

**Pull request merged**